sirepo/job_supervisor.py

```diff
@@ -691,7 +710,16 @@ def _req_is_valid(self, req):
                         await self.run_dir_acquire(o)
                     await o.prepare_send()
                     o.send()
-                    return await o.reply_get()
+                    r =  await o.reply_get()
+                    # POSIT: any api_* that could run into runDirNotFound
+                    # will call _send_with_single_reply() and this will
+                    # properly format the reply
+                    if not r.get('runDirNotFound'):
+                        return r
+                    self.__db_init(req, prev_db=self.db)
+                    assert self.db.status == job.MISSING, \
```

Comment:
This particular assertion is ok for now. I'm not sure we want it in the long run. At some point we have to trust the code to do what it is supposed to do. We aren't programming Eiffel. We have tests for this type of validation.